WooCommerce/Classes/ViewRelated/Products/Edit Product/ProductFormViewController.swift

```diff
@@ -605,7 +617,15 @@ private extension ProductFormViewController {
 //
 private extension ProductFormViewController {
     func saveProduct(status: ProductStatus? = nil) {
-        let productStatus = status ?? product.status
+        var productStatus = status ?? product.status
+
+        // Set productStatus to custom type for saving a variation
+        // This will change Progress view title and message for variation save action.
+        // Fix for the issue: https://github.com/woocommerce/woocommerce-ios/issues/4847
+        if let _ = self.viewModel.productionVariationID {
+            productStatus = .custom("variation")
+        }
```

Comment:
⚠️ I'm concerned about this, I don't think we should be changing the product status just to be able to detect if we are saving a variation or not.

You can see how in this class we detect if we are dealing with a variation with something like 
```swift
viewModel is ProductVariationFormViewModel
```

or by just checking the `productionVariationID ` property you just added.

Would you mind changing it, please?

Reply:
Sure, I will change it to `viewModel is ProductVariationFormViewModel`.